common/app/model/dotcomrendering/pageElements/PageElement.scala

```diff
@@ -1228,7 +1228,7 @@ object PageElement {
           )
           .toList
       case Interactive =>
-        element.interactiveTypeData.flatMap(_.iframeUrl).map(url => InteractiveBlockElement(url)).toList
+        element.interactiveTypeData.map(d => InteractiveBlockElement(d.iframeUrl, d.alt, d.scriptUrl, d.role, d.isMandatory)).toList
```

Comment:
Does this need to include elementId or is that already set somewhere else?

Reply:
Pretty sure that's added magically to elements at serialisation time, right @shtukas ?